var/spack/repos/builtin/packages/gatk/package.py

```diff
@@ -3,59 +3,96 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-from spack import *
 import os.path
+import glob
+from spack import *
 
 
 class Gatk(Package):
-    """Genome Analysis Toolkit
-       Variant Discovery in High-Throughput Sequencing Data
     """
+    Genome Analysis Toolkit
+    Variant Discovery in High-Throughput Sequencing Data
+    """
+
     homepage = "https://software.broadinstitute.org/gatk/"
-    url      = "https://github.com/broadinstitute/gatk/releases/download/4.0.4.0/gatk-4.0.4.0.zip"
-
-    version('4.0.11.0', sha256='5ee23159be7c65051335ac155444c6a49c4d8e3515d4227646c0686819934536')
-    version('4.0.8.1', sha256='6d47463dfd8c16ffae82fd29e4e73503e5b7cd0fcc6fea2ed50ee3760dd9acd9',
-            url='https://github.com/broadinstitute/gatk/archive/4.0.8.1.tar.gz')
-    version('4.0.4.0', '083d655883fb251e837eb2458141fc2b',
-            url="https://github.com/broadinstitute/gatk/releases/download/4.0.4.0/gatk-4.0.4.0.zip")
-    version('3.8-1', 'a0829534d2d0ca3ebfbd3b524a9b50427ff238e0db400d6e9e479242d98cbe5c', extension='tar.bz2',
-            url="https://software.broadinstitute.org/gatk/download/auth?package=GATK-archive&version=3.8-1-0-gf15c1c3ef")
-    version('3.8-0', '0581308d2a25f10d11d3dfd0d6e4d28e', extension='tar.gz',
-            url="https://software.broadinstitute.org/gatk/download/auth?package=GATK")
-
-    depends_on('java@8:', type='run')
-    depends_on('python@2.6:2.8,3.6:', type='run', when='@4.0:')
-    depends_on('r@3.2:', type='run', when='@4.0:')
+    url = "https://github.com/broadinstitute/gatk/releases/download/4.0.4.0/gatk-4.0.4.0.zip"
+
+    version(
+        "4.1.0.0",
+        sha256="148aa061328d922a570d0120d88f27e61e5da877f542206f4d77f2d788b7d21d",
+        url="https://github.com/broadinstitute/gatk/releases/download/4.1.0.0/gatk-4.1.0.0.zip",
```

Comment:
Most of these versions follow the format of the package-level URL, so you shouldn't have to specify individual URLs. `3.8-1` appears to have a different format so that would need the version-specific URL.

Reply:
I've added list_url and removed the individual urls from the 4.x versions.